tools/mo/openvino/tools/mo/ops/split.py

```diff
@@ -212,6 +252,38 @@ def infer(node):
         elif op == 'AttributedSplit':
             PermuteAttrs.create_permute_attrs(node, attrs=[('axis', 'input:0')])
 
+    @staticmethod
+    def reverse_infer(node: Node):
+        assert hasattr(node, 'num_splits')
+        axis = node.in_port(1).data.get_value() if node.op == 'Split' else node.soft_get('axis', None)
+        assert axis is not None, '{} `axis` is unknown for node {}'.format(node.op, node.soft_get('name', node.id))
+
+        shapes = []
+        splited_dim_size = 0
+        for i in range(node['num_splits']):
+            shape = node.out_port(i).data.get_shape()
+            if shape is not None:
+                shapes.append(shape)
+            else:
+                # if at least one output shape is None/undefined we cannot define
+                # exact value of shape along axis
+                splited_dim_size = dynamic_dimension
+                continue
+
+            # if input is splitted into several outputs, e.g.
+            # out_shape_1 = [dyn, 4, 3], out_shape_2 = [7, dyn, 3], axis = 2
+            # to get the original source shape [7, 4, 6]
+            # dimensions of axis = 2 must be summed while
+            # for other dimensions clarify_partial_shape should be called
+            splited_dim_size += shapes[-1][axis]
+            shapes[-1][axis] = dynamic_dimension
+
+        if len(shapes) == 0:
+            return
+        precise_shape = clarify_partial_shape(shapes)
+        precise_shape[axis] = splited_dim_size
+        node.in_port(0).data.set_shape(precise_shape)
```

Comment:
a lot of duplicated code

Reply:
implemented a singe function without duplication